packages/backend/src/clients/Slack/SlackClient.ts

```diff
@@ -369,8 +369,29 @@ export class SlackClient {
     async postFileToThread(args: PostSlackFile) {
         const webClient = await this.getWebClient(args.organizationUuid);
 
+        let { channelId } = args;
+        // If the channel ID is a user ID (starts with U), we need to open a DM channel first
+        // Slack API's files.uploadV2 doesn't support uploading files where channel_id is a user ID
+        if (channelId.startsWith('U')) {
+            try {
+                const response = await webClient.conversations.open({
```

Comment:
this call then requires updating the Slack apps scope to include im:write

wonder if we want to make this change + update all apps

**or**

if we could instead of uploading a pdf attachment, a link to the PDF file ?

Reply:
We have been moving to attachments to avoid link/s3 expiration issues.
Does updating the scopes require orgs to reinstall the app? 
Does this open a new channel each time?

Reply:
> Does updating the scopes require orgs to reinstall the app?

yes

> Does this open a new channel each time?

No, it sends from the app like a DM 

<img width="1437" alt="image" src="https://github.com/user-attachments/assets/8eddc331-bd98-445b-bea5-19e68a0236fd" />

Reply:
will do!

Reply:
Updated internal and external docs https://docs.lightdash.com/self-host/customize-deployment/configure-a-slack-app-for-lightdash#create-a-new-slack-app-on-your-organization

Set a task for tomorrow's deployment 👍